### PR TITLE
add image pull secret

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: liuwei <liuweixa@redhat.com>

### CODE REVIEW

**Comment by skeeey** [comment]
refer to https://github.com/open-cluster-management/backlog/issues/17034

File: pkg/hub/addon/addon.go [hunk 1]
```diff
@@ -81,13 +92,28 @@ func (a *clusterProxyAddOnAgent) Manifests(cluster *clusterv1.ManagedCluster, ad
 		return nil, fmt.Errorf("no data in ca-bundle-crt configmap: %v", caConfigMap)
 	}
 
-	host, port := a.anpPublicHost, a.anpPublicPort
+	imagePullSecret := os.Getenv(defaultImagePullSecretEnvVarName)
+	agentImage := a.agentImage
+	if _, ok := cluster.Labels[imageregistryv1alpha1.ClusterImageRegistryLabel]; ok {
```
[the remark this answers is not in the document]
Comment (skeeey) [reply pkg/hub/addon/addon.go]: done

File: pkg/hub/addon/addon.go [hunk 2]
```diff
@@ -81,13 +91,26 @@ func (a *clusterProxyAddOnAgent) Manifests(cluster *clusterv1.ManagedCluster, ad
...
+		return nil, err
+	}
+
+	imagePullSecret := os.Getenv(defaultImagePullSecretEnvVarName)
+	pullSecret, err := a.imageRegistryClient.Cluster(cluster.Name).PullSecret()
+	if err != nil {
+		return nil, err
+	}
+	if pullSecret != nil {
+		imagePullSecret = pullSecret.Name
```
Comment (zhiweiyin318) [inline pkg/hub/addon/addon.go:105]: pullSecret may not in the same ns with the pod

Comment (skeeey) [reply pkg/hub/addon/addon.go]: fixed to `open-cluster-management-image-pull-credentials`

**Pull request merged**